test/e2e/upgrade_hypershift_operator_test.go

```diff
@@ -0,0 +1,247 @@
+
+		operatorImage, err = e2eutil.GetHyperShiftOperatorImage(ctx, mc, globalOpts)
+		g.Expect(err).ToNot(gomega.HaveOccurred(), "Getting HyperShiftOperator image shouldn't return errors")
+		g.Expect(operatorImage).To(gomega.Equal(hyperShiftOperatorLatestImage))
+
+		t.Logf("Observed post-upgrade HyperShift Operator image %q", operatorImage)
+
+		g.Expect(t.Run("Verify upgrade invariants", func(t *testing.T) {
+			t.Log("Verifying upgrade invariants")
+			namespace := hostedCluster.Namespace
+			g := gomega.NewWithT(t)
+
+			postUpgradeNodePools := &hyperv1.NodePoolList{}
+			err = mgmtClient.List(ctx, postUpgradeNodePools, crclient.InNamespace(namespace))
+			g.Expect(err).ToNot(gomega.HaveOccurred(), "Listing NodePools in namespace %s shouldn't return errors",
+				namespace)
+
+			g.Expect(postUpgradeNodePools.Items).ToNot(gomega.BeEmpty(), "Should find NodePools in namespace %s",
+				namespace)
+			g.Expect(len(postUpgradeNodePools.Items)).To(gomega.BeEquivalentTo(len(nodePoolsMap)),
+				"Number of pre-ugrade and post-upgrade NodePools should match")
+
+			for _, nodePool := range postUpgradeNodePools.Items {
+				t.Logf("Verifying NodePool %s", nodePool.Name)
+				preUpgradeNodePool, ok := nodePoolsMap[nodePool.Name]
+				g.Expect(ok).To(gomega.BeTrue(), "Should have a pre-upgrade nodepool with the name %s",
+					nodePool.Name)
+
+				t.Logf("Generation: %d", nodePool.Generation)
+				t.Logf("CurrentConfig: %s", nodePool.Annotations[nodePoolAnnotationCurrentConfig])
+				t.Logf("CurrentConfigVersion:%s", nodePool.Annotations[nodePoolAnnotationCurrentConfigVersion])
+
+				// Check if the node pool has been updated
+				g.Expect(nodePool.Generation).To(gomega.Equal(preUpgradeNodePool.Generation),
+					"Pre-upgrade and post-upgrade NodePool generations should match")
+				g.Expect(nodePool.Annotations[nodePoolAnnotationCurrentConfig]).To(
+					gomega.Equal(preUpgradeNodePool.Annotations[nodePoolAnnotationCurrentConfig]),
+					"Pre-upgrade and post-upgrade NodePool current config should match")
+				g.Expect(nodePool.Annotations[nodePoolAnnotationCurrentConfigVersion]).To(
+					gomega.Equal(preUpgradeNodePool.Annotations[nodePoolAnnotationCurrentConfigVersion]),
+					"Pre-upgrade and post-upgrade NodePool current config version should match")
+
+				conditions, err := e2eutil.Conditions(&nodePool)
+				g.Expect(err).ToNot(gomega.HaveOccurred(), "Retrieving NodePool conditions shouldn't return errors")
+				targetConditions := sets.NewString(hyperv1.NodePoolUpdatingVersionConditionType,
+					hyperv1.NodePoolUpdatingConfigConditionType, hyperv1.NodePoolUpdatingPlatformMachineTemplateConditionType)
+				for _, c := range conditions {
+					if targetConditions.Has(c.Type) {
+						t.Logf("Found condition %s of NodePool %s", c.String(), nodePool.Name)
+						g.Expect(c.Status).NotTo(gomega.Equal(metav1.ConditionTrue), "Condition %s of nodepool %s shouldn't be True", nodePool.Name, c.Type)
```

Comment:
nit: we know the expected state should bit false.

Reply:
Fixed